key.ui/src/main/java/de/uka/ilkd/key/gui/MainWindow.java

```diff
@@ -1707,12 +1710,9 @@ public synchronized void autoModeStopped(ProofEvent e) {
             getMediator().addKeYSelectionListenerChecked(proofListener);
         }
 
-        /**
-         * invoked when the strategy of a proof has been changed
-         */
         @Override
-        public synchronized void settingsChanged(EventObject e) {
-            if (proof.getSettings().getStrategySettings() == e.getSource()) {
+        public void propertyChange(PropertyChangeEvent evt) {
+            if (proof.getSettings().getStrategySettings() == evt.getSource()) {
```

Comment:
`proof` is accessed and `synchronized` was removed. Is this right? See the Github Annotation above.